### DISCUSSION TITLE
Differences in AR1 spatiotemporal model

### DISCUSSION THREAD

**Discussion by seananderson** [vast-lib/tinyVAST#44] [Q&A] [answered]
Is there an obvious reason why these models are slightly different? Perhaps because of exactly how the AR1 is constructed or the AR1 correlation parameter going into sdmTMB as a parameter internally transformed to -1 to 1 (`ar1_phi`)?

I'm not ruling out that there's a longstanding bug in sdmTMB. However, I've checked the current sdmTMB implementation against a 

```
 SCALE(SEPARABLE(AR1(rho(m)), GMRF(Q_temp, s)), 1./exp(ln_tau_E(m)))(epsilon_st.col(m));
```

implementation and I get identical answers. Seems to be different with or without the extra time elements. IID spatiotemporal fields match.

I'm happy to do more legwork digging into why they differ, but if there's an obvious reason I'll stop here.

``` r
library(sdmTMB)
library(tinyVAST)

mesh <- make_mesh(pcod, c("X", "Y"), cutoff = 20)
fit_sdmTMB <- sdmTMB(
  density ~ 1,
  data = pcod, mesh = mesh,
  family = tweedie(link = "log"),
  time = "year",
  extra_time = seq(min(pcod$year), max(pcod$year)),
  spatiotemporal = "ar1",
  spatial = "off"
)
fit_sdmTMB$sd_report
#> sdreport(.) result
#>            Estimate Std. Error
#> b_j       2.7596742 0.26985383
#> ln_tau_E -0.1163046 0.45658085
#> ln_kappa -2.1344361 0.29388459
#> thetaf    0.2729091 0.04374530
#> ln_phi    2.6345224 0.03138427
#> ar1_phi   2.4316957 0.29181597
#> Maximum gradient component: 5.43726e-09
ps1 <- as.list(fit_sdmTMB$sd_report, "Estimate")
ps2 <- as.list(fit_sdmTMB$sd_report, "Estimate", report = TRUE)
ps <- c(ps1, ps2)

pcod$var <- "logn"
fit_tv <- tinyVAST::tinyVAST(
  density ~ 1,
  dsem = "logn -> logn, 1, rho",
  data = pcod,
  time_column = "year",
  times = seq(min(pcod$year), max(pcod$year)),
  space_columns = c("X", "Y"),
  spatial_graph = mesh$mesh,
  family = tweedie(),
  control = tinyVASTcontrol(newton_loops = 1)
)
fit_tv$sdrep
#> sdreport(.) result
#>             Estimate Std. Error
#> alpha_j    2.8999068 0.21837650
#> beta_z     0.8230102 0.04920501
#> beta_z     1.6873453 0.34455984
#> log_sigma  2.6354518 0.03138101
#> log_sigma  0.2682081 0.04351482
#> log_kappa -2.0748597 0.30015390
#> Maximum gradient component: 1.038297e-09
pt <- as.list(fit_tv$sdrep, "Estimate")

# Close, but not exact...
pt$beta_z[1]
#> [1] 0.8230102
ps$rho
#> [1] 0.8384251

ps$b_j
#> [1] 2.759674
pt$alpha_j
#> [1] 2.899907

ps$thetaf
#> [1] 0.2729091
pt$log_sigma[2]
#> [1] 0.2682081

ps$ln_phi
#> [1] 2.634522
pt$log_sigma[1]
#> [1] 2.635452

pt$beta_z[2]
#> [1] 1.687345
ps$sigma_E[1,1]
#> [1] 2.678424
```

<sup>Created on 2024-05-02 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

**Comment by James-Thorson-NOAA** [comment, marked as answer]
Yeah, tinyVAST uses the DSEM machinery to roll up the functionality of DFA, ARIMA, etc.  However, I haven't derived the stationary distribution in this super-general case, and so it starts the variance at the conditional (not the marginal, a.k.a., stationary) value.  So for AR1+ models, there will be mismatch relative to using `density::AR1(rho)`.  And it's been a while since I checked the initial variance for epsilon in VAST or sdmTMB :0  Anyhoo, no big surprise, and glad that it's small ... I hope to eventually derive that stationary variance and add that as an option.

**Comment by seananderson** [comment]
Awesome, yeah, I was wondering if it was that. I guess the biggest difference being in the sigma_E/beta_z[2] is consistent with that. Otherwise, everything I've tested so far has matched, including Poisson-link delta models. sdmTMB makes a manual adjustment to match the stationary AR1 distribution (so it can be built up with a time-varying sigma) and I'm always paranoid I've messed that up.
